cleanlab/multilabel_classification/filter.py

```diff
@@ -134,10 +134,8 @@ def find_label_issues(
             warnings.warn(f"`rank_by_kwargs` is not used when `low_memory=True`.")
         arg_values = {
             "filter_by": filter_by,
-            "frac_noise": frac_noise,
```

Comment:
I think we should still check if users pass in non-default args for this, my previous comment was just pointing out that the default arg is not `None`, so we need to check for it slightly differently

Reply:
Got it. How would we check if the argument has a default or non-default value? The only way I can think of is

```
if frac_noise ! = 1.0:
    warnings.warn("..")
if min_examples_per_class != 1:
   warnings.warn("..")
```
This doesn't seem right though, because if the default values are changed, then they would need to be changed in these conditions as well.  But if this works, I can make the relevant changes.

Reply:
Since cleanlab does use the `inspect` module at certain places, this might also work.
```python
import inspect
signature = inspect.signature(find_label_issues)
default_args =  {
    k: v.default
    for k, v in signature.parameters.items()
    if v.default is not inspect.Parameter.empty
}
arg_values = {
    "filter_by": filter_by,
    "num_to_remove_per_class": num_to_remove_per_class,
    "confident_joint": confident_joint,
    "n_jobs": n_jobs,
    "num_to_remove_per_class": num_to_remove_per_class,
    "frac_noise": frac_noise,
    "min_examples_per_class": min_examples_per_class
}
# If an argument does not have a default value or if 
# passed value differs from the default, raise warning, 
for arg_name, arg_val in arg_values.items():
    if arg_name not in default_args or arg_val != default_args[arg_name]:
        warnings.warn(f"`{arg_name}` is not used when `low_memory=True`.")
```
Adapted from this [SO answer](https://stackoverflow.com/a/12627202/8505509).

Reply:
I like the `inspect` approach! 

One suggestion - I believe you can simplify the last if statement to just this?

```
for arg_name, arg_val in arg_values.items():
    if arg_val != default_args[arg_name]:
        print(f"`{arg_name}` is not used when `low_memory=True`.")
```

I'm not entirely sure what `arg_name not in default_args` is checking for since I think arguments with `None` values will still be in the `default_args` dict. Please double check this tho and correct me if I'm wrong / am missing something.

Reply:
You are right! I will simplify the condition, thanks!
As a side note, I observed that I made similar changes in other places where the `low_memory` option has been introduced. In the future, it might make sense to make a common function `check_args(arg_values, func)` that uses the signature of the function and specified arguments to prints warnings.